networkx/algorithms/community/modularity_max.py

```diff
@@ -234,7 +215,98 @@ def greedy_modularity_communities(G, weight=None, resolution=1, n_communities=1)
             b[v] += b[u]
             b[u] = 0
 
-    return sorted(communities.values(), key=len, reverse=True)
+        yield list(communities.values())
+
+
+def greedy_modularity_communities(
+    G, weight=None, resolution=1, cutoff=1, best_n=None, **aliases
+):
+    """
+    Parameters
+    ----------
+    G : NetworkX graph
+
+    weight : string or None, optional (default=None)
+        The name of an edge attribute that holds the numerical value used
+        as a weight.  If None, then each edge has weight 1.
+        The degree is the sum of the edge weights adjacent to the node.
+
+    resolution : float, optional (default=1)
+        If resolution is less than 1, modularity favors larger communities.
+        Greater than 1 favors smaller communities.
+
+    cutoff : int or None, optional (default=1)
+        The number of communities in the partition at which the merging process
+        will stop even if a maximum is not yet reached.
+
+    best_n : int or None, optional (default=None)
+        Force community merging to continue until `best_n` communities remain
+        even if the modularity decreases.
+        If ``None``, don't force it to continue beyond a maximum.
+
+    Returns
+    -------
+    communities: list
+        A list of frozensets of nodes, one for each community.
+        Sorted by length with largest communities first.
+
+    Examples
+    --------
+    >>> from networkx.algorithms.community import greedy_modularity_communities
+    >>> G = nx.karate_club_graph()
+    >>> c = greedy_modularity_communities(G)
+    >>> sorted(c[0])
+    [8, 14, 15, 18, 20, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33]
+
+    See Also
+    --------
+    modularity
+    """
+    if (cutoff < 1) or (cutoff > G.number_of_nodes()):
+        raise ValueError(
+            f"``cutoff`` must be between 1 and {G.number_of_nodes()}. Got {cutoff}."
+        )
+    if best_n is not None:
+        if (best_n < 1) or (best_n > G.number_of_nodes()):
+            raise ValueError(
+                f"``best_n`` must be between 1 and {G.number_of_nodes()}. Got {best_n}."
+            )
+        if best_n < cutoff:
+            raise ValueError(
+                f"``best_n`` (got {best_n}) must be greater or equal than ``cutoff`` "
+                f"(got {cutoff})."
+            )
+    else:
+        best_n = G.number_of_nodes()
```

Comment:
Slightly shorter code in the formatting strings allows `black` to play nice.
```suggestion
    if (cutoff < 1) or (cutoff > G.number_of_nodes()):
        raise ValueError(f"best_n must be between 1 and {len(G)}. Got {best_n}.)"
    if best_n is not None:
        if (best_n < 1) or (best_n > G.number_of_nodes()):
            raise ValueError(f"best_n must be between 1 and {len(G)}. Got {best_n}.")
        if best_n < cutoff:
            raise ValueError(f"Must have best_n >= cutoff. Got {best_n} < {cutoff}")
    else:
        best_n = G.number_of_nodes()
```